packages/lodestar/src/network/network.ts

```diff
@@ -99,6 +100,39 @@ export class Libp2pNetwork extends (EventEmitter as { new(): NetworkEventEmitter
     await this.libp2p.hangUp(peerInfo);
   }
 
+  // peerIds: peers already have this subnet
+  public async connectToNewPeersBySubnet(subnet: number, inPeerIds?: string[]): Promise<number> {
+    const peerIds = inPeerIds || [];
+    // @ts-ignore
+    const discovery: Discv5Discovery = this.libp2p._discovery.get("discv5");
+    const discv5: Discv5 = discovery.discv5;
+    const peerInfosForSubnet: PeerInfo[] = await Promise.all(
+      discv5.kadValues()
+        .filter((enr: ENR) => enr.get("attnets"))
+        .filter((enr: ENR) => this.config.types.AttestationSubnets.deserialize(enr.get("attnets"))[subnet])
```

Comment:
we cannot control what peers will put here, so we should catch malformed deserialization errors